src/bin/ion/commands/beta/generate/templates/rust/struct.templ

```diff
@@ -6,68 +6,75 @@ use crate::ion_data_model::{{ import.name | snake }}::{{ import.name | upper_cam
                                             {{ target_kind_name }}::read_from(reader)?;
                                         {% else %}
                                             reader.read_{{ target_kind_name }}()?;
                                         {% endif %}
-        {% else %}
-            {% if abstract_data_type == "Struct"%}reader.step_in()?;{% endif %}
-            let mut abstract_data_type = {{ target_kind_name }}::default();
+        {% elif abstract_data_type is object and abstract_data_type | get(key="Structure") %}
+            reader.step_in()?;
             while reader.next()? != StreamItem::Nothing {
                 if let Some(field_name) = reader.field_name()?.text() {
                     match field_name {
                         {% for field in fields -%}
                             {% if field.value | is_built_in_type == false %}
-                                {% if abstract_data_type is object and abstract_data_type["Sequence"] | length != 0 %}
-                                 "{{ field.name }}" => { abstract_data_type.{{ field.name }} = {
-                                     let mut values = vec![];
-                                     reader.step_in()?;
-                                     while reader.next()? != StreamItem::Nothing {
-                                        {% if abstract_data_type["Sequence"] | is_built_in_type == false %}
-                                            values.push({{ abstract_data_type["Sequence"] }}::read_from(reader)?);
-                                        {% else %}
-                                            values.push(reader.read_{{ abstract_data_type["Sequence"] }}()?);
-                                        {% endif %}
-                                    }
-                                    values
-                                 };}
-                                 {% elif abstract_data_type == "Struct" %}
-                                 "{{ field.name }}" => { abstract_data_type.{{ field.name }} = {{ field.value }}::read_from(reader)?; }
-                                 {% endif %}
+                                 "{{ field.name }}" => { abstract_data_type.{{ field.name | snake }} = {{ field.value }}::read_from(reader)?; }
                             {% else %}
-                            "{{ field.name }}" => { abstract_data_type.{{ field.name }} = reader.read_{{ field.value }}()?; }
+                                "{{ field.name }}" => { abstract_data_type.{{ field.name | snake}} = reader.read_{{ field.value | lower }}()?; }
                             {% endif %}
                         {% endfor %}
-                     _ => {}
+                     _ => {
+                        {% if abstract_data_type["Structure"] %}
+                            return IonResult::decoding_error(
+                                "Can not read field name:{{ field.name }} for {{ target_kind_name }} "
```

Comment:
Can you make the error message a little more specific? Why can't the field name be read? Is it an unknown field? Is it the wrong type? Is it unreadable because it's missing?

Reply:
Sure. I will add that change.